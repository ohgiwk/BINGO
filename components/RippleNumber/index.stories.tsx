import React from 'react'
import RippleNumber from './'

const config = {
  title: 'RippleNumber',
}
export default config

export const defaultRippleNumber = () => (
  <RippleNumber
    {...{
      open: false,
      value: '0',
      ripple: false,
    }}
  />
)
export const openRippleNumber = () => (
  <RippleNumber
    {...{
      open: false,
      value: '0',
      ripple: false,
    }}
  />
)
