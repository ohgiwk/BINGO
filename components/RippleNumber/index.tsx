import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import classNames from 'classnames'

/**
 *
 * @param props
 * @returns
 */
const RippleNumber: React.FC<{
  value: string
  open: boolean
  ripple: boolean
  className?: string
}> = (props) => {
  const classes = useStyles()
  return (
    <div
      className={classNames(props.className, classes.num, classes.ripple, {
        [`${classes.onRipple}`]: props.open && props.ripple,
        [`${classes.open}`]: props.open,
      })}
    >
      {props.value}
    </div>
  )
}
export default RippleNumber

const useStyles = makeStyles((theme) => ({
  num: {
    fontSize: '2rem',
    color: 'gray',
    width: '40px',
    height: '40px',
    borderRadius: '50%',
    lineHeight: '40px',
    textAlign: 'center',
    userSelect: 'none',
    transition: 'color 0.5s, transform 0.4s',
    '&:hover': {
      transform: 'scale(1.4)',
    },
  },
  open: { color: theme.palette.primary.main },
  ripple: {
    position: 'relative',
    '&:after': {
      content: '""',
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
      background: theme.palette.primary.main,
      borderRadius: '50%',
      width: 0,
      height: 0,
      opacity: 1,
      transition: 'opacity 1s, width 1s, height 1s',
    },
  },
  onRipple: {
    '&:after': {
      content: '',
      width: '150px',
      height: '150px',
      opacity: 0,
      transition: 'opacity 1s, width 1s, height 1s',
      pointerEvents: 'none',
    },
  },
}))
