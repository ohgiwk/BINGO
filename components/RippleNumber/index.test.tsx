import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import RippleNumber from '.'

describe('RippleNumber', () => {
  test('Snapshot Test', () => {
    const props = {
      open: false,
      value: '0',
      ripple: false,
    }

    const component = renderer.create(<RippleNumber {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('ラベルテキストが正しく表示されているか確認', () => {
    const props = {
      open: false,
      value: '0',
      ripple: false,
    }

    // propsを受け取り mount テスト対象component生成
    const wrapper = mount(<RippleNumber {...props} />)

    // { value }にpropsで受け取った値が表示されているか確認
    expect(wrapper.find('div').text()).toBe(props.value)
  })
})
