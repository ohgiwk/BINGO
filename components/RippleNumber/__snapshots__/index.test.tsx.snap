// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RippleNumber Snapshot Test 1`] = `
<div
  className="
        makeStyles-num-1
        
        makeStyles-ripple-3
        
        
      "
>
  0
</div>
`;
